### PR TITLE
Use std::size_t for indexes and sizes instead of unsigned int

### PR DESCRIPTION
**Pull request by victorlevasseur**
Use std::size_t for indexes and sizes instead of unsigned int (unsigned int is possibly lossy for storing sizes and indexes on 64 bits systems).

**Not really tested, however, it should not cause bugs as, for most cases (numbers < 2^32), the conversion between unsigned int and std::size_t is lossless.**

### CODE REVIEW

**Comment by 4ian** [comment]
This should fix the warnings concerning cast from size_t to unsigned int creating potential losses of precisions (even if this won't happen in practice).  :)
I'll probably merge this after merging the quite huge changes here: https://github.com/4ian/GD/tree/feature/behavior-renaming

**Comment by victorlevasseur** [comment]
This should be ok now. :)

**Comment by victorlevasseur** [comment]
This should really be ok now ^^.

**Comment by victorlevasseur** [comment]
By the way, many of these loops could be avoided by using iterators.

**Comment by 4ian** [comment]
Compiled, no more warning about casting from size_t to unsigned int with potential loss of precision :+1: 
Did you do a global find/replace on unsigned int or a manual change on a case by case basis?

File: GDCpp/GDCpp/BuiltinExtensions/ProfileTools.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-/*
+/*std::size_t
```
Comment (4ian) [inline GDCpp/GDCpp/BuiltinExtensions/ProfileTools.cpp]: @victorlevasseur Tiny mistake here ahah :smile:

**Comment by victorlevasseur** [comment]
Typo fixed :+1:

**Comment by victorlevasseur** [comment]
I almost did a global find/replace but still watch the result so as it stays consistent, e.g. don't use std::size_t instead of unsigned int for things that do not represent size of index (there are not a lot).

**Comment by 4ian** [comment]
Ok that's indeed the best approach :) Merging

**Pull request merged**